### DISCUSSION TITLE
"No hparams data was found" in tensorboard

### DISCUSSION THREAD

**Discussion by adosar** [Lightning-AI/pytorch-lightning#19776] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
According to the [docs](https://lightning.ai/docs/pytorch/stable/extensions/logging.html#logging-hyperparameters):

> If you want to track a metric in the tensorboard hparams tab, log scalars to the key `hp_metric`.

I am logging the `hp_metric` in the end of validation epoch like this:

```python
def on_validation_epoch_end(self):
    preds = torch.cat(self.val_step_preds)
    targets = torch.cat(self.val_step_targets)
    
    metric = self.metric(preds=preds, target=targets)
    
    self.log('hp_metric', metric)
    self.log('val_metric', metric, prog_bar=True)
```
When I visualize my logs with `tensorboard` under the `HPARAMS` tab I get: "No hparams data was found".

Any ideas?
